packages/purpose-process/src/services/purposeService.ts

```diff
@@ -734,6 +734,340 @@ export function purposeServiceBuilder(
       return newPurposeVersion;
     },
 
+    async activatePurposeVersion({
+      purposeId,
+      versionId,
+      organizationId,
+      correlationId,
+      logger,
+    }: {
+      purposeId: PurposeId;
+      versionId: PurposeVersionId;
+      organizationId: TenantId;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<PurposeVersion> {
+      logger.info(`Activating Version ${versionId} in Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+      const purposeVersion = retrievePurposeVersion(versionId, purpose);
+
+      const eservice = await retrieveEService(
+        purpose.data.eserviceId,
+        readModelService
+      );
+
+      if (purposeVersion.state === purposeVersionState.draft) {
+        const riskAnalysisForm = purpose.data.riskAnalysisForm;
+
+        if (!riskAnalysisForm) {
+          throw missingRiskAnalysis(purposeId);
+        }
+
+        const tenant = await getInvolvedTenantByEServiceMode(
+          eservice,
+          purpose.data.consumerId,
+          readModelService
+        );
+
+        assertTenantKindExists(tenant);
```

Comment:
@Carminepo2 check [this](https://github.com/pagopa/interop-be-monorepo/pull/476/files#diff-95b4e078c342343517e86a98ea5650ede6c4ec6e5c9f4547dbfb44c8d6d22118R986-R996). I know the branches are not in straight sequence, so that function isn't available here, but we should keep the same pattern. Let's wait for who merges first and then adjust this. cc @AsterITA